### PR TITLE
chore(react-grid): merge fixed types and names

### PR DESCRIPTION
**Pull request by kvet**

### CODE REVIEW

File: packages/dx-grid-core/src/plugins/table-row-detail/constants.js [hunk 1]
```diff
@@ -1 +1 @@
-export const TABLE_DETAIL_TYPE = 'detail';
+export const TABLE_DETAIL_TYPE = Symbol('detail');
```
Comment (MaximKudriavtsev) [inline packages/dx-grid-core/src/plugins/table-row-detail/constants.js]: Why we don't use `.toString()` here?

Comment (kvet) [reply packages/dx-grid-core/src/plugins/table-row-detail/constants.js]: We use toString only in the key definition. Type should be symbol.

File: packages/dx-react-grid-demos/src/demo-sources/grid-fixed-columns/with-selection.jsxt [hunk 2]
```diff
@@ -65,9 +63,8 @@ export default class Demo extends React.PureComponent {
           <TableHeaderRow />
           <TableSelection />
           <TableFixedColumns
-            leftColumnTypes={leftColumnTypes}
-            leftColumnNames={leftColumnNames}
-            rightColumnNames={rightColumnNames}
+            leftColumns={leftColumns}
+            rightColumns={rightColumns}
```
Comment (MaximKudriavtsev) [inline packages/dx-react-grid-demos/src/demo-sources/grid-fixed-columns/with-selection.jsxt]: The sentence `You can also use the leftColumnTypes and rightColumnTypes properties to fix service columns, such as Selection Column or Editing Column.` is keeping in the doc

Comment (kvet) [reply packages/dx-react-grid-demos/src/demo-sources/grid-fixed-columns/with-selection.jsxt]: Fixed

**Pull request merged**